test/mock-server.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2017, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const nock = require('nock');
+
+const config = require('config');
+const constants = require('./../lib/constants');
+const fixtureUser = require('./fixtures/user.json');
+const fixtureApps = require('./fixtures/apps.json');
+const fixtureApp = require('./fixtures/app.json');
+const fixtureServices = require('./fixtures/datalinks.json');
+const fixtureJob = require('./fixtures/job.json');
+const fixtureInternalDataLink = require('./fixtures/kinvey-dlc.json');
+const fixtureLogs = require('./fixtures/logs.json');
+
+/**
+ * Mocks our MAPI server. Serves as a wrapper around nock.js.
+ */
+class MockServer {
+  /**
+   * @param requireAuth If true, all endpoints (except those for login) would check for user token and respond with 4xx if such is not present.
```

Comment:
Well, I'm all in for some kind of code documentation. It doesn't take much time when writing the code and saves some afterwards. Maybe, it isn't needed everywhere, probably it could be skipped for some obvious things but otherwise it seems useful to me. (Just wanted to share my thoughts while we're waiting for Mike). 
:smiley: